### DISCUSSION TITLE
Browse Spotify library in my own build

### DISCUSSION THREAD

**Discussion by ads5and** [BimmerGestalt/AAIdrive#727] [Q&A] [answered]
Hello, thank you for developing such a useful application. :)

I wanted to make some changes to the app, but I'm having trouble with browsing the Spotify library in my own build. The official build from Play Store has no problems with the integration of Spotify on my smartphone (Samsung S10).

In my build the Spotify App is listed in the app list, but with the information that it can only be controlled during usage. I was able to tick the search option and get the Web API authorization, but I'm unable to tick the "browse Spotify library". In the player overlay in AAIDrive I get the UserNotAutorizedException, with the info that the user has to complete the auth-flow to use Spotify on their behalf.

What could be the potential causes of the problem that I could troubleshoot?
Should I look for some specific properties in my build? Whether the reason could be something related to the API Key, fingerprint or other factors related to the authorization?
Thank you for any hints about this.

**Comment by hufman** [comment, marked as answer]
Good morning, and thank you for wanting to help contribute!
How have you set up the Spotify developer key? Perhaps the [Android package fingerprint](https://developer.spotify.com/documentation/android/tutorials/application-fingerprints) is needed, unlike the documentation which only hints that it's helpful.

![image](https://github.com/BimmerGestalt/AAIdrive/assets/1592375/f7d750ac-e8a1-4515-b8f7-2d0fe100e1ee)

**Reply by ads5and** [reply]
Thank you for your reply. It turns out that I had a problem with the fingerprint. I can also confirm that in my case the fingerprint was necessary to browse the library.

**Reply by hufman** [reply]
Did it work without any fingerprint at all? Or it was definitely required?

**Reply by ads5and** [reply]
Browsing the library didn't work without fingerprint. Only the search option was working. So it seems that it is required.

**Reply by hufman** [reply]
Thank you for the confirmation, I'll update the docs!
